nntrainer/tensor/tensor_pool.h

```diff
@@ -340,18 +299,35 @@ class TensorPool {
                         unsigned int offset);
 
 private:
+  /**
+   * @brief Source tensor detailed specification
+   *
+   */
+  struct SourceDetails {
```

Comment:
for better naming, how about TensorDetails and DepTensorDetatils? looks like Source* isn't proper for this. How about make object like class or structure having these details.. and treating with it?